lib/jumubase_web/controllers/performance_controller.ex

```diff
@@ -0,0 +1,46 @@
+defmodule JumubaseWeb.PerformanceController do
+  use JumubaseWeb, :controller
+  alias Ecto.Changeset
+  alias Jumubase.Foundation
+  alias Jumubase.Foundation.Contest
+  alias Jumubase.Showtime
+  alias Jumubase.Showtime.{Appearance, Participant, Performance}
+
+  # Pass contest from nested route to all actions
+  def action(conn, _), do: get_contest!(conn, __MODULE__)
+
+  def new(conn, _params, contest) do
+    changeset =
+      %Performance{appearances: [%Appearance{participant: %Participant{}}]}
+      |> Showtime.change_performance()
```

Comment:
This seems to always creates a changeset with a "contest category can't be blank" error. We could try pre-filling the `contest_category_id` to the first category in the list.

Reply:
Hm, but with the Vue-less form I never saw that error on the field before submitting… Not sure if I want to have it pre-filled, as it would be good to have users consciously pick one (remember many of them are lazy teenagers 😉 and there is no automatic way to detect wrong category submissions)

Reply:
I haven't studied your added code enough yet, but isn't it so that changeset errors are still displayed in the form only after submitting? In that case, it seems weird that the `contest_category_id` error shows while the empty participant fields have no effect, doesn't it?

Reply:
The code simply shows errors if the changeset has any. I didn't think that any detection of whether there was a previous submit would be required. How would we check this? By inspecting the current route?

Reply:
You're right, the changeset created in the lines quoted above does indeed have only a single error (missing `contest_category_id`), nothing about the nested `Appearance` and `Performance` associations even though those are equally incomplete…

Reply:
https://hexdocs.pm/phoenix_html/Phoenix.HTML.Form.html:

> If no action has been applied to the changeset or action was set to :ignore, no errors are shown on the form object even if the changeset has a non-empty :errors value.

> This is useful for things like validation hints on form fields, e.g. an empty changeset for a new form. That changeset isn’t valid, but we don’t want to show errors until an actual user action has been performed.

Reply:
Sounds good, but I just looked at the changeset returned by `Showtime.create_performance` and its `action` is still `nil`. If we can make it return `:create` instead, we could use `action` to decide whether to serialize `errors`.

Reply:
Ah, I think I broke that myself 😬 Have a look at L44 in `showtime.ex` – I added some code to check if the CC matches the contest from the nested route… and of course an empty CC id doesn't match that. So the changeset we're getting doesn't actually come back from `Repo` at all.

Reply:
Ah ok. We just can set `action` manually in that case.

Reply:
Managed to fix it on my machine :) However, I'm now considering doing the whole "does selected CC match contest" check a little differently, e.g. as an extra validation step on the changeset before `Repo.insert` that only happens when the changeset is otherwise valid. I don't really want to see a "CC is not in given contest" error for an empty CC id anyway 😛

Reply:
Hehe, true. 😆 Anyway, I just pushed an immediate fix (probably the same one you have), everything should be working now.